store/ethereum.go

```diff
@@ -0,0 +1,23 @@
+package store
+
+import (
+	"math/big"
+	"time"
+)
+
+type IncomingEthereumTransaction struct {
+	Hash               string
+	ValueWei           *big.Int
+	StellarAddress     string
+	WithdrawExpiration time.Time
+
+	TransactionBlob string
+}
+
+func (m *Memory) GetIncomingEthereumTransactionByHash(hash string) (IncomingEthereumTransaction, error) {
```

Comment:
We should not need to ingest ethereum transactions. So I think it makes more sense to include this function in a struct representing the ethereum client instead of the Memory store